factory/processing/blockProcessorCreator.go

```diff
@@ -459,7 +457,12 @@ func (pcf *processComponentsFactory) newShardBlockProcessor(
 	}, nil
 }
 
-func (pcf *processComponentsFactory) addSystemVMToContainer(vmContainer process.VirtualMachinesContainer, builtInFuncFactory vmcommon.BuiltInFunctionFactory) error {
+func (pcf *processComponentsFactory) addSystemVMToContainerIfNeeded(vmContainer process.VirtualMachinesContainer, builtInFuncFactory vmcommon.BuiltInFunctionFactory) error {
```

Comment:
We should generally not have functions like that `ifNeeded`. Would've been better to keep the if on the outside at caller side, but I understand we need to keep this for testing purposes. We need to create a task for this in JIRA so it doesn't get lost.

Reply:
yes, it will be refactored with the runTypeComponents and Creator

Reply:
added jira task